src/controllers/rolePresetsController.js

```diff
@@ -1,25 +1,29 @@
-const { hasPermission } = require('../utilities/permissions');
+const helper = require('../utilities/permissions');
 
 const rolePresetsController = function (Preset) {
   const getPresetsByRole = async function (req, res) {
-    if (!(await hasPermission(req.body.requestor, 'putRole'))) {
-      res.status(403).send('You are not authorized to make changes to roles.');
+    if (!(await helper.hasPermission(req.body.requestor, 'putRole'))) {
+      res.status(403).send({ error: 'You are not authorized to make changes to roles.' });
       return;
     }
 
     const { roleName } = req.params;
     Preset.find({ roleName })
       .then((results) => {
-        res.status(200).send(results);
+        if (results.length === 0) {
```

Comment:
I actually like the fact that you added this piece of logic. I might still impact the frontend, since you are using an object rather than a string as a response?

Also, if the results array is  of length zero, is it required to return an error or should we just leave it as an empty array and the frontend will figure out what to do with it?

In short, is change due to a request by Jae or any other stakeholder?